frontend/src/pages/RecipientRecord/pages/MergeGoals/index.js

```diff
@@ -0,0 +1,385 @@
+        return false;
+      }
+      return true;
+    },
+    message: 'At least 2 goals must be selected.',
+  },
+};
+
+const steps = [
+  {
+    position: SELECT_GOALS_TO_MERGE,
+    label: 'Select goals to merge',
+  },
+  {
+    position: SELECT_GOALS_TO_KEEP,
+    label: 'Select goal to keep',
+  },
+  {
+    position: REVIEW_AND_MERGE,
+    label: 'Review merged goal',
+  },
+];
+
+const stepIndicatorStatus = (position, activePage) => {
+  if (position === activePage) {
+    return 'current';
+  }
+
+  if (position < activePage) {
+    return 'complete';
+  }
+
+  return 'incomplete';
+};
+
+export const navigate = (newPage, setActivePage) => {
+  if (!pages.includes(newPage)) {
+    return;
+  }
+
+  setActivePage(newPage);
+  window.scrollTo(0, 0);
+};
+
+export default function MergeGoals({
+  location,
+  recipientId,
+  regionId,
+  recipientNameWithRegion,
+  canMergeGoals,
+}) {
+  const [error, setError] = useState('');
+  const [validation, setValidation] = useState('');
+  const [goals, setGoals] = useState([]);
+  const [activePage, setActivePage] = useState(SELECT_GOALS_TO_MERGE);
+  const { setIsAppLoading } = useContext(AppLoadingContext);
+  const drawerTriggerRef = useRef(null);
+
+  const hookForm = useForm({
+    mode: 'onSubmit',
+    defaultValues: {
+      selectedGoalIds: [],
+      finalGoalId: null,
+    },
+  });
+
+  const { register, watch } = hookForm;
+  const selectedGoalIds = watch('selectedGoalIds');
+  const finalGoalId = watch('finalGoalId');
+
+  useEffect(() => {
+    async function fetchGoals() {
+      const goalIds = new URLSearchParams(location.search).getAll('goalId[]');
+
+      if (!goalIds.length) {
+        setError('No goal ids provided');
+        return;
+      }
+
+      try {
+        setError('');
+        setIsAppLoading(true);
+        // this might seem a weird place for this but I've found that
+        // putting it in its own useEffect or even outside of this function
+        // causes this message to briefly be displayed
+        // we put it here so that it gets swept up in the rendering passes
+        // without ever being visible
+        if (!canMergeGoals) {
+          setError('You do not have permission to merge goals for this recipient');
+        }
+        const { goalRows } = await getRecipientGoals(
+          recipientId,
+          regionId,
+          'goal',
+          'asc',
+          OFFSET,
+          false,
+          false,
+          goalIds,
+        );
+        setGoals(goalRows);
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        console.error(e);
+        setError('Unable to fetch goals');
+      } finally {
+        // remove loading screen
+        setIsAppLoading(false);
+      }
+    }
+
+    fetchGoals();
+  }, [location.search, recipientId, regionId, setIsAppLoading, canMergeGoals]);
+
+  const selectedGoals = goals.filter((g) => (
+    selectedGoalIds.includes(g.id.toString())
+  ));
```

Comment:
If I only have one goal in my database and I try to "review 1 similar goal", `selectedGoalIds` flips between being a number, an array, and a boolean. I'm guessing this will never happen in the real world so maybe we really don't care about this, but thought I'd mention it anyways.

```javascript
  const selectedGoals = goals.filter((g) => {
    console.log('selectedGoalIds is', selectedGoalIds);
    return selectedGoalIds.includes(g.id.toString());
  });
```

![Screenshot 2023-10-27 at 9 04 51 AM](https://github.com/HHS/Head-Start-TTADP/assets/17074357/74c818c4-b171-4b93-99a0-83699c49ccc0)

When clicking on a goal card checkbox on the /merge route:

![Screenshot 2023-10-27 at 9 05 56 AM](https://github.com/HHS/Head-Start-TTADP/assets/17074357/18c86ec9-622b-4f47-89db-52fb410e7e83)

Feel free to just resolve this if you think it's a non-issue 🙃